extensions/azurecore/src/azurecore.d.ts

```diff
@@ -355,7 +355,22 @@ declare module 'azurecore' {
 			getTreeDataProvider(): IAzureResourceTreeDataProvider;
 		}
 
-		export interface IAzureResourceTreeDataProvider extends TreeDataProvider<IAzureResourceNode> {
+		export interface IAzureResourceTreeDataProvider {
+			 /**
+			  * Gets the root tree item nodes for this provider - these will be used as
+			  * direct children of the Account node in the Azure tree view.
+			  */
+			getRootChildren(): Promise<azdata.TreeItem[]>;
```

Comment:
Doing this of course breaks the "contract" we had - but as far as I know there aren't any other extensions that contribute Azure resources outside of azurecore so that shouldn't be a problem. I'll do a quick run through of our extensions to make sure though before getting this in.